generators/kubernetes/templates/istio/gateway.yml.ejs

```diff
@@ -33,13 +33,13 @@ spec:
       name: http
       protocol: HTTP
     hosts:
-    - <%= app.baseName.toLowerCase() %>.<%= kubernetesNamespace %><%= ingressDomain %>
+    - <%= app.baseName.toLowerCase() %>.<%= ingressDomain %>
```

Comment:
I see no harm in having the host name like this (without K8s namespace). Since this will be deployed to a specific namespace, 'host' would get resolved automatically. 

But, as you know, in general, services (normal/headless) in k8s (without any modification) get assigned a default DNS 'A' name record of the form {service_name}.{namespace}.svc.{internal_domain_name}. So '<%= kubernetesNamespace %>' as part of the host look-up is in-line with the default naming convention. Same logic is applicable in all other places as well. 

To resolve an external domain properly, add a wild-card 'A' name record for the base domain. For example if the base domain is, let's say jhip-k8s.jhipster.com, then the 'A' name record should be 
*. jhip-k8s.jhipster.com   A    <LB_IP>
This would resolve all the services properly like (jaeger.istio-system.jhip-k8s.jhipster.com, account.space1.jhip-k8s.jhipster.com etc.)

Reply:
Thanks, Srini. Maybe since I was using a wild card DNS, I encountered this issue. So can I merge this? also, the original template had a missing `.`  between `<%= kubernetesNamespace %>` & `<%= ingressDomain %>` right?

Reply:
Refer this - It's already taken care of in the logic.

https://github.com/jhipster/generator-jhipster/blob/65fc5d40650d355d6176a7c6aa58b707e1757c09/generators/kubernetes/prompts.js#L113


I would prefer to leave the naming convention and mapping the way it is today.

Reply:
Ok thats why it didn't work when I used JDL, I thinks its safer to do the `.` concatenation in the template as its clearer to understand and will work for all cases. I'll fix it and try again

Reply:
@srinivasa-vasu you said this change should also work right? in that case can we keep it as its simpler especially when using wild card DNS

Reply:
@deepu105 The existing code should work with wildcard DNS. Are you facing any specific issue?

Reply:
For me, existi8ng code didn't work, also for this guy https://medium.com/@simionrazvan/microservices-using-jhipster-kubernetes-istio-bf75e992612a

Reply:
Srini, I'm merging this for the moment and will open another PR to introduce the namespace back in the URLs